### ISSUE TITLE
Script doesn't work with raspberry pi os bullseye.

### ISSUE THREAD

**Issue by RaptorHunterMZ** [sdetweil/MagicMirror_scripts#36]
The script doesn't work with the newest version of raspberry pi os.

**Comment by sdetweil** [comment]
can u attach the ~/install.log

**Comment by bert269** [comment]
Sorry - not in ~ or in MagicMirror subdir. Where  else can I look.
I did see this during the install: 
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@es-joy/jsdoccomment@0.10.8',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16' },
npm WARN EBADENGINE   current: { node: 'v17.1.0', npm: '8.1.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'eslint-plugin-jsdoc@36.1.0',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16' },
npm WARN EBADENGINE   current: { node: 'v17.1.0', npm: '8.1.2' }
npm WARN EBADENGINE }
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.

**Comment by bert269** [comment]
Here you go
[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7529837/install.log)
.

**Comment by BKeyport** [comment]
[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7531135/install.log)

**Comment by sdetweil** [comment]
try updated script 
```
bash -c  "$(curl -sL https://raw.githubusercontent.com/sdetweil/MagicMirror_scripts/bullseye/raspberry.sh)"
```

**Comment by BKeyport** [comment]
install log being saved to /home/pi/install.log
Installing helper tools ...
Updating packages ...
[     <=>                                                                                                                                       ]bash: line 151: 0: command not found
bash: line 155: node: command not found
bash: line 156: [: -lt: unary operator expected
Check current Node installation ...
Node.js is not installed.
Installing Node.js ...
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages were automatically installed and are no longer required:
  libc-ares2 libjs-highlight.js nodejs-doc
Use 'sudo apt autoremove' to remove them.
The following NEW packages will be installed:
  nodejs
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/22.9 MB of archives.
After this operation, 116 MB of additional disk space will be used.
(Reading database ... 99829 files and directories currently installed.)
Preparing to unpack .../nodejs_14.18.1-deb-1nodesource1_armhf.deb ...
Unpacking nodejs (14.18.1-deb-1nodesource1) ...
dpkg: error processing archive /var/cache/apt/archives/nodejs_14.18.1-deb-1nodesource1_armhf.deb (--unpack):
 trying to overwrite '/usr/share/doc/nodejs/api/embedding.json.gz', which is also in package nodejs-doc 12.22.5~dfsg-2~11u1
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/nodejs_14.18.1-deb-1nodesource1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
node version is bash: line 253: node: command not found
bash: line 255: node: command not found
Node.js installation Done! version=
Check current NPM installation ...
NPM currently installed. Checking version number.
/usr/bin/env: ‘node’: No such file or directory
Minimum npm version: V6.0.0
Installed npm version: V
npm should be upgraded.
Installing npm ...
E: Unable to correct problems, you have held broken packages.
/usr/bin/env: ‘node’: No such file or directory
/usr/bin/env: ‘node’: No such file or directory
npm installation Done! version=V
Cloning MagicMirror ...
Cloning into 'MagicMirror'...
remote: Enumerating objects: 325, done.
remote: Counting objects: 100% (325/325), done.
remote: Compressing objects: 100% (290/290), done.
remote: Total 325 (delta 53), reused 146 (delta 22), pack-reused 0
Receiving objects: 100% (325/325), 477.94 KiB | 1.63 MiB/s, done.
Resolving deltas: 100% (53/53), done.
Cloning MagicMirror Done!
Installing dependencies ...
/usr/bin/env: ‘node’: No such file or directory
Unable to install dependencies!

**Comment by BKeyport** [comment]
[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7534728/install.log)

**Comment by sdetweil** [comment]
think latest resolves this

**Comment by Nexx1c** [comment]
I think I'm having the same issue:

> install starting  - Fri Nov 19 13:57:21 GMT 2021
installing on armv7l processor system
the os is Distributor ID: Raspbian Description: Raspbian GNU/Linux 11 (bullseye) Release: 11 Codename: bullseye
Installing helper tools ...
Reading package lists...
Building dependency tree...
Reading state information...
build-essential is already the newest version (12.9).
curl is already the newest version (7.74.0-1.3).
git is already the newest version (1:2.30.2-1).
unzip is already the newest version (6.0-26).
wget is already the newest version (1.21-1rpi1).
wget set to manually installed.
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 nodejs : Depends: libnode72 (= 12.22.5~dfsg-2~11u1) but it is not going to be installed
Updating packages ...
Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease Hit:3 https://deb.nodesource.com/node_14.x bullseye InRelease Reading package lists...
apt-get update  completed ok
apt-get upgrade  started
apt upgrade result =rc=127
Check current Node installation ...
Node currently installed. Checking version number.
forcing low Node version
Minimum Node version: v14.0.0
Installed Node version: V1.0.0
Node should be upgraded.
Installing Node.js ...
Node release info = ## Installing the NodeSource Node.js 14.x repo... ## Populating apt-get cache... + apt-get update Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease Hit:3 https://deb.nodesource.com/node_14.x bullseye InRelease Reading package lists... ## Confirming "bullseye" is supported... + curl -sLf -o /dev/null 'https://deb.nodesource.com/node_14.x/dists/bullseye/Release' ## Adding the NodeSource signing key to your keyring... + curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee /usr/share/keyrings/nodesource.gpg >/dev/null ## Creating apt sources list file for the NodeSource Node.js 14.x repo... + echo 'deb [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_14.x bullseye main' > /etc/apt/sources.list.d/nodesource.list + echo 'deb-src [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_14.x bullseye main' >> /etc/apt/sources.list.d/nodesource.list ## Running `apt-get update` for you... + apt-get update Hit:1 http://archive.raspberrypi.org/debian bullseye InRelease Hit:2 http://raspbian.raspberrypi.org/raspbian bullseye InRelease Hit:3 https://deb.nodesource.com/node_14.x bullseye InRelease Reading package lists... ## Run `sudo apt-get install -y nodejs` to install Node.js 14.x and npm ## You may also need development tools to build native addons: sudo apt-get install gcc g++ make ## To install the Yarn package manager, run: curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor | sudo tee /usr/share/keyrings/yarnkey.gpg >/dev/null echo "deb [signed-by=/usr/share/keyrings/yarnkey.gpg] https://dl.yarnpkg.com/debian stable main" | sudo tee /etc/apt/sources.list.d/yarn.list sudo apt-get update && sudo apt-get install yarn
Node.js installation Done! version=
Check current NPM installation ...
npm is not installed.
Installing npm ...
Reading package lists...
Building dependency tree...
Reading state information...
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 nodejs : Depends: libnode72 (= 12.22.5~dfsg-2~11u1) but it is not going to be installed
 npm : Depends: node-gyp (>= 3.6.2~) but it is not going to be installed
upgrading npm to latest
npm installation Done! version=V
Cloning MagicMirror ...
Cloning MagicMirror Done!
Installing dependencies ...
Unable to install dependencies!


When I try installing nodejs manually (sudo apt install -y nodejs) I get the following error message:
> Unpacking nodejs (14.18.1-deb-1nodesource1) over (12.22.5~dfsg-2~11u1) ...
dpkg: error processing archive /var/cache/apt/archives/nodejs_14.18.1-deb-1nodesource1_armhf.deb (--unpack):
 trying to overwrite '/usr/share/doc/nodejs/api/embedding.json.gz', which is also in package nodejs-doc 12.22.5~dfsg-2~11u1
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/nodejs_14.18.1-deb-1nodesource1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

Running on a brand new Raspi 4 with Bullseye.

**Comment by sdetweil** [comment]
@Nexx1c you are not running the bullseye version of the script.. there are two  just for testing

**Comment by Nexx1c** [comment]
Yepp, that was definitely an oversigt on my part..
So is raspberry.sh updated for Bullseye, or is it still just in testing? 
I tried with the raspberry.sh script, but got the same issue:

> installing on armv7l processor system
the os is Distributor ID: Raspbian Description: Raspbian GNU/Linux 11 (bullseye) Release: 11 Codename: bullseye
Installing helper tools ...
Reading package lists...
Building dependency tree...
Reading state information...
build-essential is already the newest version (12.9).
curl is already the newest version (7.74.0-1.3).
git is already the newest version (1:2.30.2-1).
unzip is already the newest version (6.0-26).
wget is already the newest version (1.21-1).
wget set to manually installed.
The following package was automatically installed and is no longer required:
  libfuse2
Use 'sudo apt autoremove' to remove it.
Suggested packages:
  doc-base
The following NEW packages will be installed:
  pv
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 46.4 kB of archives.
After this operation, 113 kB of additional disk space will be used.
Get:1 http://raspbian.playstar.se/raspbian bullseye/main armhf pv armhf 1.6.6-1 [46.4 kB]
Fetched 46.4 kB in 0s (101 kB/s)
Selecting previously unselected package pv.
(Reading database ... 170981 files and directories currently installed.)
Preparing to unpack .../archives/pv_1.6.6-1_armhf.deb ...
Unpacking pv (1.6.6-1) ...
Setting up pv (1.6.6-1) ...
Processing triggers for man-db (2.9.4-2) ...
Updating packages ...
Get:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease [15.0 kB] Get:2 http://archive.raspberrypi.org/debian bullseye InRelease [23.6 kB] Get:3 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf Packages [13.2 MB] Get:4 http://archive.raspberrypi.org/debian bullseye/main armhf Packages [207 kB] Fetched 13.5 MB in 8s (1,585 kB/s) Reading package lists...
apt-get update  completed ok
apt-get upgrade  started
apt upgrade result =rc=0 Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 vlc-bin : Depends: libvlc-bin (= 3.0.16-1+rpi1+rpt1) but 3.0.16-1+rpi1+rpt2 is to be installed
 vlc-plugin-skins2 : Depends: vlc-plugin-qt (= 3.0.16-1+rpi1+rpt2) but 3.0.16-1+rpi1+rpt1 is to be installed
E: Broken packages
Check current Node installation ...
Node currently installed. Checking version number.
Minimum Node version: v14.0.0
Installed Node version: v12.22.5
Node should be upgraded.
Installing Node.js ...
Node release info = ## Installing the NodeSource Node.js 14.x repo... ## Populating apt-get cache... + apt-get update Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease Reading package lists... ## Confirming "bullseye" is supported... + curl -sLf -o /dev/null 'https://deb.nodesource.com/node_14.x/dists/bullseye/Release' ## Adding the NodeSource signing key to your keyring... + curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee /usr/share/keyrings/nodesource.gpg >/dev/null ## Creating apt sources list file for the NodeSource Node.js 14.x repo... + echo 'deb [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_14.x bullseye main' > /etc/apt/sources.list.d/nodesource.list + echo 'deb-src [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_14.x bullseye main' >> /etc/apt/sources.list.d/nodesource.list ## Running `apt-get update` for you... + apt-get update Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease Get:3 https://deb.nodesource.com/node_14.x bullseye InRelease [4,586 B] Get:4 https://deb.nodesource.com/node_14.x bullseye/main armhf Packages [781 B] Fetched 5,367 B in 2s (3,228 B/s) Reading package lists... ## Run `sudo apt-get install -y nodejs` to install Node.js 14.x and npm ## You may also need development tools to build native addons: sudo apt-get install gcc g++ make ## To install the Yarn package manager, run: curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor | sudo tee /usr/share/keyrings/yarnkey.gpg >/dev/null echo "deb [signed-by=/usr/share/keyrings/yarnkey.gpg] https://dl.yarnpkg.com/debian stable main" | sudo tee /etc/apt/sources.list.d/yarn.list sudo apt-get update && sudo apt-get install yarn
Node.js installation Done! version=
Check current NPM installation ...
npm is not installed.
Installing npm ...
Reading package lists...
Building dependency tree...
Reading state information...
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 nodejs : Depends: libnode72 (= 12.22.5~dfsg-2~11u1) but it is not going to be installed
 npm : Depends: node-gyp (>= 3.6.2~) but it is not going to be installed
upgrading npm to latest
npm installation Done! version=V
Cloning MagicMirror ...
Cloning MagicMirror Done!
Installing dependencies ...
Unable to install dependencies!

**Comment by sdetweil** [comment]
still not the right script.. same name, different branch..  still in test.. only a couple users have tested...

you do not clone my repo, you just execute from there..   copy/paste the one line 
with bullseye in the url.. 

note that if u try the non-bullseye  version  you end up with a mess you cannot get out of..  new flash  is it..

**Comment by Nexx1c** [comment]
Alright, that's... quite embarrassing.. Let's see if I got it right this time.. -_-

So this time I ran the raspberry.sh from the bullseye branch:
 bash -c  "$(curl -sL https://raw.githubusercontent.com/sdetweil/MagicMirror_scripts/bullseye/raspberry.sh)"

And things started looking better, but with some errors during the install, and no ~/MagicMirror folder created..

>pi@raspberrypi:~ $ cat install.log
install starting  - Sat Nov 20 11:22:59 GMT 2021
installing on armv7l processor system
the os is Distributor ID: Raspbian Description: Raspbian GNU/Linux 11 (bullseye) Release: 11 Codename: bullseye
Installing helper tools ...
Reading package lists...
Building dependency tree...
Reading state information...
build-essential is already the newest version (12.9).
curl is already the newest version (7.74.0-1.3).
git is already the newest version (1:2.30.2-1).
unzip is already the newest version (6.0-26).
wget is already the newest version (1.21-1).
wget set to manually installed.
The following package was automatically installed and is no longer required:
  libfuse2
Use 'sudo apt autoremove' to remove it.
Suggested packages:
  doc-base
The following NEW packages will be installed:
  pv
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 46.4 kB of archives.
After this operation, 113 kB of additional disk space will be used.
Get:1 http://raspbian.playstar.se/raspbian bullseye/main armhf pv armhf 1.6.6-1 [46.4 kB]
Fetched 46.4 kB in 0s (104 kB/s)
Selecting previously unselected package pv.
(Reading database ... 170981 files and directories currently installed.)
Preparing to unpack .../archives/pv_1.6.6-1_armhf.deb ...
Unpacking pv (1.6.6-1) ...
Setting up pv (1.6.6-1) ...
Processing triggers for man-db (2.9.4-2) ...
Updating packages ...
Get:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease [15.0 kB] Get:2 http://archive.raspberrypi.org/debian bullseye InRelease [23.6 kB] Get:3 http://raspbian.raspberrypi.org/raspbian bullseye/main armhf Packages [13.2 MB] Get:4 http://archive.raspberrypi.org/debian bullseye/main armhf Packages [207 kB] Fetched 13.5 MB in 11s (1,279 kB/s) Reading package lists...
apt-get update  completed ok
apt-get upgrade  started
apt upgrade result =rc=0 Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
 vlc-bin : Depends: libvlc-bin (= 3.0.16-1+rpi1+rpt1) but 3.0.16-1+rpi1+rpt2 is to be installed
 vlc-plugin-skins2 : Depends: vlc-plugin-qt (= 3.0.16-1+rpi1+rpt2) but 3.0.16-1+rpi1+rpt1 is to be installed
E: Broken packages
package.json update for armv7l failed
setting up initial config.js
Check plymouth installation ...
Splashscreen: Checking themes directory.
Splashscreen: Create theme directory if not exists.
Splashscreen: Copying theme failed!
install and setup pm2
pm2 not installed, installing
pm2 install result npm WARN deprecated uuid@3.4.0: Please upgrade to version 7 or higher. Older versions may use Math.random() in certain circumstances, which is known to be problematic. See https://v8.dev/blog/math-random for details. added 181 packages, and audited 182 packages in 23s 10 packages are looking for funding run `npm fund` for details found 0 vulnerabilities
get the pm2 platform specific startup command
startup command = sudo env PATH=$PATH:/usr/bin:/bin /usr/local/lib/node_modules/pm2/bin/pm2 startup systemd -u pi --hp /home/pi
__/\\\\\\\\\\\\\____/\\\\____________/\\\\____/\\\\\\\\\_____
 _\/\\\/////////\\\_\/\\\\\\________/\\\\\\__/\\\///////\\\___
  _\/\\\_______\/\\\_\/\\\//\\\____/\\\//\\\_\///______\//\\\__
   _\/\\\\\\\\\\\\\/__\/\\\\///\\\/\\\/_\/\\\___________/\\\/___
    _\/\\\/////////____\/\\\__\///\\\/___\/\\\________/\\\//_____
     _\/\\\_____________\/\\\____\///_____\/\\\_____/\\\//________
      _\/\\\_____________\/\\\_____________\/\\\___/\\\/___________
       _\/\\\_____________\/\\\_____________\/\\\__/\\\\\\\\\\\\\\\_
        _\///______________\///______________\///__\///////////////__

                        Runtime Edition
        PM2 is a Production Process Manager for Node.js applications
                     with a built-in Load Balancer.
                Start and Daemonize any application:
                $ pm2 start app.js
                Load Balance 4 instances of api.js:
                $ pm2 start api.js -i 4
                Monitor in production:
                $ pm2 monitor
                Make pm2 auto-boot at server restart:
                $ pm2 startup
               To go further checkout:
                http://pm2.io/
>[PM2] Init System found: systemd
Platform systemd
Template
[Unit]
Description=PM2 process manager
Documentation=https://pm2.keymetrics.io/
After=network.target
[Service]
Type=forking
User=pi
LimitNOFILE=infinity
LimitNPROC=infinity
LimitCORE=infinity
Environment=PATH=$PATH:/usr/bin:/bin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
Environment=PM2_HOME=/home/pi/.pm2
PIDFile=/home/pi/.pm2/pm2.pid
Restart=on-failure
ExecStart=/usr/local/lib/node_modules/pm2/bin/pm2 resurrect
ExecReload=/usr/local/lib/node_modules/pm2/bin/pm2 reload all
ExecStop=/usr/local/lib/node_modules/pm2/bin/pm2 kill
[Install]
WantedBy=multi-user.target
Target path
/etc/systemd/system/pm2-pi.service
Command list
[ 'systemctl enable pm2-pi' ]
[PM2] Writing init configuration in /etc/systemd/system/pm2-pi.service
[PM2] Making script booting at startup...
[PM2] [-] Executing: systemctl enable pm2-pi...
[PM2] [v] Command successfully executed.
+---------------------------------------+
[PM2] Freeze a process list on reboot via:
$ pm2 save
[PM2] Remove init script via:
$ pm2 unstartup systemd
pm2 startup command done
configure the pm2 config file for MagicMirror
start MagicMirror via pm2 now
save MagicMirror pm2 config now
stop MagicMirror via pm2 now
disable screensaver via gsettings was true and uint32 300
disable screensaver via lightdm.conf
disable screensaver via lxsession
We're ready! Run pm2 start MagicMirror from the ~/MagicMirror directory to start your MagicMirror.
install completed - Sat Nov 20 11:24:13 GMT 2021

**Comment by sdetweil** [comment]
did u start w a fresh SD card flash? it's a mess otherwise.

**Comment by sdetweil** [comment]
the readme online says

if doing bullseye use this link

**Comment by Nexx1c** [comment]
Yes, fresh SD card flash each time. 

Yeah, I used that link in the last couple of attempts. 

Seems like there's quite a few broken packages with bullseye, so I went back to buster (which brought it's own suite of issues since it's oldstable, but at least it seems to be working now.

**Comment by pingywon** [comment]
Yeah it is true - this script is  broke on a fresh install of bullseye. I flashed 2x - brand new installs of bullseye and got a ton of errors after running the scri
[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7599567/install.log)
pt


I was hoping for an easy install...

**Comment by sdetweil** [comment]
there is an updated script for bullseye.. ready the instructions again

**Comment by sdetweil** [comment]
@pingywon   see my reply

**Comment by Wolfieeewolf** [comment]
looks like new script runs with errors (see install.log)

getting error on cmd 

bash: line 398: package.json: No such file or directory

cp: cannot stat '/home/pi/MagicMirror/splashscreen/splash.png': No such file or directory
Splashscreen: Copying theme failed!

bash: line 522: installers/pm2_MagicMirror.json: No such file or directory
bash: line 523: installers/mm.sh: No such file or directory
chmod: cannot access 'installers/mm.sh': No such file or directory
[PM2][ERROR] File /home/pi/MagicMirror/installers/pm2_MagicMirror.json not found
[PM2] Saving current process list...
[PM2][WARN] PM2 is not managing any process, skipping save...
[PM2][WARN] To force saving use: pm2 save --force

when i run " pm2 start MagicMirror" i get this [PM2][ERROR] Script not found: /home/pi/MagicMirror




[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7600412/install.log)

**Comment by sdetweil** [comment]
the log does not have the special message about installing on bullseye. 

u flashed the SD card
booted the sf card
and the ran my script by copy/paste from the GitHub page, correct? and u used the bullseye url

**Comment by pingywon** [comment]
> there is an updated script for bullseye.. ready the instructions again
> looking at the log, you used the old script

Can you link me to the correct script because your English is very unclear. I used the script that is front and center on your git hub here that says it’s for bullseye. I do not know what “ready the instructions again” means.

**Comment by sdetweil** [comment]
@pingywon  'ready' was a typo, should have been 'read'

this is the text I reference

```
if you are installing on the latest release bullseye, use this url (while we are testing) , it will also work on buster or any other system too
bash  -c  "$(curl -sL https://raw.githubusercontent.com/sdetweil/MagicMirror_scripts/bullseye/raspberry.sh)"
```

the output of the log does not have the sentence
```
echo "installing on bullseye' >>$logfile
```
which is written only to the log

**Comment by pingywon** [comment]
That is what I have tried 2x. 

I now reflashed a third time to bullseye from the pi imager app. I will run the script again and report back but I’m here to tell you that there is indeed fatal errors in the script. At the end it keeps trying to do things in the MagicMirror dir and that dir isn’t even created.  Do I need to do a git pull from mitchmitch before running your script?

**Comment by sdetweil** [comment]
no, the script does everything

**Comment by sdetweil** [comment]
I need to see the ~/install.log file

**Comment by pingywon** [comment]
Alright. I figured it should. Like I said I have a brand new install of bullseye flashed. I’m not even gonna run updates. Just gonna run the script and will report back. Thank you

**Comment by pingywon** [comment]
Script is broke. Fresh flash of bullseye. Here is the log:


`Linux MagicMirror 5.10.63-v7+ #1459 SMP Wed Oct 6 16:41:10 BST 2021 armv7l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Thu Nov 25 23:12:01 2021 from 192.168.1.238
pi@MagicMirror:~ $ ls
4  4#  Bookshelf  Desktop  Documents  Downloads  install.log  Music  Pictures  Public  Templates  Videos
pi@MagicMirror:~ $ cat install.log | more

install starting  - Thu Nov 25 23:17:40 EST 2021
installing on armv7l processor system
the os is Distributor ID: Raspbian Description: Raspbian GNU/Linux 11 (bullseye) Release: 11 Codename: bullseye
Installing helper tools ...
Reading package lists...
Building dependency tree...
Reading state information...
build-essential is already the newest version (12.9).
curl is already the newest version (7.74.0-1.3).
git is already the newest version (1:2.30.2-1).
unzip is already the newest version (6.0-26).
The following package was automatically installed and is no longer required:
  libfuse2
Use 'sudo apt autoremove' to remove it.
Suggested packages:
  doc-base
The following NEW packages will be installed:
  pv
The following packages will be upgraded:
  wget
1 upgraded, 1 newly installed, 0 to remove and 77 not upgraded.
Need to get 968 kB of archives.
After this operation, 113 kB of additional disk space will be used.
Get:1 http://mirror.us.leaseweb.net/raspbian/raspbian bullseye/main armhf wget armhf 1.21-1rpi1 [922 kB]
Get:2 http://mirror.us.leaseweb.net/raspbian/raspbian bullseye/main armhf pv armhf 1.6.6-1 [46.4 kB]
apt-listchanges: Reading changelogs...
Fetched 968 kB in 2s (575 kB/s)
(Reading database ... 
Preparing to unpack .../wget_1.21-1rpi1_armhf.deb ...
Unpacking wget (1.21-1rpi1) over (1.21-1) ...
Selecting previously unselected package pv.
Preparing to unpack .../archives/pv_1.6.6-1_armhf.deb ...
Unpacking pv (1.6.6-1) ...
Setting up wget (1.21-1rpi1) ...
Setting up pv (1.6.6-1) ...
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for install-info (6.7.0.dfsg.2-6) ...
Updating packages ...
Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease Reading package lists...
apt-get update  completed ok
apt-get upgrade  started
apt upgrade result =rc=0 Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 vlc-bin : Depends: libvlc-bin (= 3.0.16-1+rpi1+rpt1) but 3.0.16-1+rpi1+rpt2 is to be installed
 vlc-plugin-skins2 : Depends: vlc-plugin-qt (= 3.0.16-1+rpi1+rpt2) but 3.0.16-1+rpi1+rpt1 is to be installed
E: Broken packages
installing on bullseye
installing correct version of node and npm, please wait
  installing : node-v14.18.1
       mkdir : /usr/local/n/versions/node/14.18.1
       fetch : https://nodejs.org/dist/v14.18.1/node-v14.18.1-linux-armv7l.tar.xz
node failed to install, exiting
pi@MagicMirror:~ $   ^C
pi@MagicMirror:~ $ 
`

**Comment by sdetweil** [comment]
ok, don't know what is going on.. 
I just flashed an sd card.. booted, skipped updates, turned on ssh, rebooted
connected via ssh, and copy/pasted the line from the scripts github for bullseye and got this 

[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7609245/install.log)
everything worked as expected event the vlc errors

**Comment by pingywon** [comment]
Well first and foremost THANK YOU for being so responsive on this thread.

Ok so I just ran the script a second time on the same install that it failed on above......and POOF it seems to be working..so strange

**Comment by sdetweil** [comment]
my pleasure.. I have opened an issue on the n github repo to see if there is a way to understand what the node install failure source is. I still suspect an intermittent networking problem..

**Comment by sdetweil** [comment]
@pingywon the n team replied that I wasn't capturing all the possible info during the node install.

if u have another SD card, and time. could u try the install again as I've update the script

**Comment by pingywon** [comment]
> @pingywon the n team replied that I wasn't capturing all the possible info during the node install.
> 
> if u have another SD card, and time. could u try the install again as I've update the script

Awesome! Yes I will reflash and run it again in the next day or so. Thanks for all the hard work and upkeep you do on this script. I first installed MM about 3-4 years ago and man as a first timer it was a long process, you have stream lined all that work very well and I think it opens the idea of a MM build to many more people. I'll report back after I reflash and run the one liner.

**Comment by pingywon** [comment]
Took me longer than I would have liked. I was waiting on a new SD card just to rule it out as the one I was using before was very old and had some heavy use. ANYWAY.

Still errors on another fresh install of Bullsey
[install.log](https://github.com/sdetweil/MagicMirror_scripts/files/7638445/install.log)
e

**Comment by sdetweil** [comment]
ok, let me look again.  there should have been more log info.....thanks for the log!!!

**Comment by pingywon** [comment]
No problem. Now that I have the new card I will try again.

**Comment by sdetweil** [comment]
I just added another redirect to capture the npm i errors into the log

**Comment by pingywon** [comment]
> I just added another redirect to capture the npm i errors into the log

Cool. I will reflash and run again tonight.

**Comment by pingywon** [comment]
> I just added another redirect to capture the npm i errors into the log

Just ran it as promised on a fresh bullseye and it was successful... I am confused, but glad.

**Comment by sdetweil** [comment]
>confused 

me too. I didn't change any functional code

only added 2>&1

to the npm i
for MagicMirror

**Comment by sdetweil** [comment]
merged bullseye fix to master
